lib/avtp_pipeline/tl/openavb_talker_no_endpoint.c

```diff
@@ -74,12 +74,14 @@ bool openavbTLRunTalkerInit(tl_state_t *pTLState)
 	pTalkerData->streamID.uniqueID = pCfg->stream_uid;
 	if (pCfg->sr_class == SR_CLASS_A) {
 		pTalkerData->classRate = 8000;
-		pTalkerData->vlanID = SR_CLASS_A_DEFAULT_VID;
+		pTalkerData->vlanID = pCfg->vlan_id == UINT16_MAX ?
+					SR_CLASS_A_DEFAULT_VID : pCfg->vlan_id;
 		pTalkerData->vlanPCP = SR_CLASS_A_DEFAULT_PRIORITY;
 	}
 	else if (pCfg->sr_class == SR_CLASS_B) {
 		pTalkerData->classRate = 4000;
-		pTalkerData->vlanID = SR_CLASS_B_DEFAULT_VID;
+		pTalkerData->vlanID = pCfg->vlan_id == UINT16_MAX ?

```

Comment:
This is a little confusing? I assume the intent is that UINT16_MAX is used as VLAN_NULL?
It would be clearer to have
#define VLAN_NULL UINT16_MAX
in a header file and then use VLAN_NULL in the source code.
Or, you could add a comment here to say that UINT16_MAX is used to represent a NULL value.